pkg/common/resource.go

```diff
@@ -81,6 +83,37 @@ func GetNodeResource(nodeStatus *v1.NodeStatus) *si.Resource {
 	return getResource(nodeStatus.Allocatable)
 }
 
+// parse cpu and memory from string to si.Resource, both of them are optional
+// if parse failed with some errors, log the error and ignore the resource
+func ParseResource(cpuStr, memStr string) *si.Resource {
+	if cpuStr == "" && memStr == "" {
+		return nil
+	}
+
+	result := NewResourceBuilder()
+	if cpuStr != "" {
+		if vcore, err := resource.ParseQuantity(cpuStr); err == nil {
+			result.AddResource(CPU, vcore.MilliValue())
+		} else {
+			log.Logger.Error("failed to parse cpu resource",
+				zap.String("cpuStr", cpuStr),
+				zap.Error(err))
```

Comment:
If we fail to parse we should not return a resource object but a nil to clearly show that we're not enforcing the quota. Returning half the quota set is going to give inconsistent behaviour.

Negative values and zeros do not have to stop parsing at this point.

Reply:
To make this be more consistent with the yunikorn-core, I think we can do that. Just updated the code for this. Thanks.